### PR TITLE
Setup ibm-watsonx-saas-da

### PR DESCRIPTION
**Pull request by andreainnocenti**
### Description

First pull request to setup the ibm-watsonx-saas-da 

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`0.3.0`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Added unit test and fixed linting issues

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by andreainnocenti** [comment]
please review @vburckhardt @ocofaigh

File: versions.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+terraform {
```
Comment (vburckhardt) [inline versions.tf:1]: I'd suggest to keep the name of the file to version.tf (without s) for overall consistency across the DAs / modules

File: .terraform-docs.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+formatter: "markdown"
```
Comment (vburckhardt) [inline .terraform-docs.yaml:1]: terradoc is already run as part of the pre-commit hooks with configuration common from https://github.com/terraform-ibm-modules/common-dev-assets/tree/8405107509b351b346887674d06ee37e98140493 , so this file is likely not needed here
Is this file necessary?

File: configure_project/variables.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
+variable "project_name" {
+  description = "Name of the Watson project to create"
```
Comment (vburckhardt) [inline configure_project/variables.tf:2]: I'd suggest to put more detailed description of the variable overall (giving some more hint on how the object that are refered to are used) - not necessarily in this PR but in the future. @SirSpidey may be able to help review the wording, etc

File: configure_project/versions.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.5.0"
```
Comment (vburckhardt) [inline configure_project/versions.tf:2]: Just set required_version = ">= 1.5.0, <1.7.0" . This is to prevent from using hashicorp tf 1.7 and above as the DA won't be tested with these versions in pipeline.

Comment (vburckhardt) [reply configure_project/versions.tf:2]: This applies across the module

File: configure_user/main.tf [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  sensitive_tokendata = sensitive(data.ibm_iam_auth_token.tokendata.iam_access_token)
+}
+
+resource "null_resource" "configure_user" {
+
+  triggers = {
+    always_run = timestamp()
+  }
+
+  provisioner "local-exec" {
```
Comment (vburckhardt) [inline configure_user/main.tf:13]: We'll need one to do the reverse action on destroy - no necessary in this PR but in future

File: configure_user/scripts/add_user.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
```
Comment (vburckhardt) [inline configure_user/scripts/add_user.sh:2]: Add set -e to force the tf execution to fail if the script error out.

File: configure_user/scripts/enforce_account_restriction.sh [hunk 7]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
```
Comment (vburckhardt) [inline configure_user/scripts/enforce_account_restriction.sh:2]: same as above on set -e

**Review by vburckhardt — COMMENTED** [review]
Hello, I put some small comments

File: configure_user/main.tf (hunk 5 above)
Comment (andreainnocenti) [reply configure_user/main.tf:13]: This is something to discuss with watson team. I don't have a strong opinion on this but IMHO since the user can have multiple ML instances, if I delete the profile, I may cause issues to other projects\ML instances.

I agree to do in another PR in case

File: configure_project/variables.tf (hunk 3 above)
Comment (andreainnocenti) [reply configure_project/variables.tf:2]: sure, @SirSpidey please help

File: configure_user/scripts/enforce_account_restriction.sh (hunk 7 above)
Comment (andreainnocenti) [reply configure_user/scripts/enforce_account_restriction.sh:2]: done

File: configure_user/scripts/add_user.sh (hunk 6 above)
Comment (andreainnocenti) [reply configure_user/scripts/add_user.sh:2]: done

File: configure_project/versions.tf (hunk 4 above)
Comment (andreainnocenti) [reply configure_project/versions.tf]: done

File: .terraform-docs.yaml (hunk 2 above)
Comment (andreainnocenti) [reply .terraform-docs.yaml]: done, I agree.

File: versions.tf (hunk 1 above)
Comment (andreainnocenti) [reply versions.tf]: done

**Review by vburckhardt — APPROVED** [review]
lgtm for merge - perhaps create issues to track remaining discussions to close from this PR
